nixos/modules/services/networking/tailscale-cert.nix

```diff
@@ -0,0 +1,143 @@
+{ config, options, lib, pkgs, ... }:
+
+let
+  cfg = config.services.tailscale-cert;
+  opt = options.services.tailscale-cert;
+in
+{
+  options.services.tailscale-cert = {
+    enable = lib.mkEnableOption "the Tailscale cert service";
+
+    package = lib.mkPackageOption pkgs "tailscale" { };
+
+    domain = lib.mkOption {
+      type = lib.types.str;
+      description = "Tailnet domain for which to request the certificate";
+      example = "monitoring.tailnet0123.ts.net";
+    };
+
+    dataDir = lib.mkOption {
+      type = lib.types.str;
+      default = "tailscale-cert";
+      description = ''
+        Directory name under /var/lib/ where to store certificates and keys.
+      '';
+    };
```

Comment:
Maybe we could just make this a `lib.types.path` and set the default to `/var/lib/tailscale-cert`? This way we don't enforce the prefix.

EDIT: not sure how this works with StateDirectory usage below

Reply:
Yeah exactly, I wanted to use `StateDirectory` for automatic state/perms management, and allowing any path here would break out of that.